File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//定义service层代码 发起http请求与后端交互
app.service("brandService",function ($http) {
    //查询所有
    this.findAll=function () {
        return  $http.get("../brand/findAll.do");
    }

    this.findPage=function (pageNum,pageSize) {
        return $http.get("../brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize);
    }

    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }

    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }

    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }

    this.search=function (pageNum,pageSize,searchEntity) {
        return $http.post('../brand/search.do?pageNum='+pageNum+"&pageSize="+pageSize,searchEntity);
    }

    this.selectBrandOptions=function () {
        return $http.get("../brand/selectBrandOptions.do");
    }

})

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有品牌列表
     * @return
     */
    public List<TbBrand> findAll();

    PageResult findPage(int pageNum, int pageSize);

    void add(TbBrand brand);

    TbBrand findOne(Long id);

    void update(TbBrand brand);

    void delete(Long[] ids);

    public PageResult search(int pageNum,int pageSize,TbBrand brand);

    /**
     * 查询模板关联的品牌下拉列表
     */
    public List<Map> selectBrandOptions();
}
